File: fixtures/scale/python/large/cipherlab/checks/check_ringshift.py
from cipherlab.ciphers.ringshift import ringshift_encrypt, ringshift_decrypt
from cipherlab.specs.ringshift_spec import ringshift_spec
from cipherlab.corpus.corpus_rotate import corpus_rotate


def check_ringshift():
    spec = ringshift_spec()
    for text in corpus_rotate():
        sealed = ringshift_encrypt(text, spec["key"])
        opened = ringshift_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "rotate"

File: fixtures/scale/ruby/large/lib/check_ringshift.rb
require_relative 'ringshift_encrypt'
require_relative 'ringshift_decrypt'
require_relative 'corpus_rotate'

def check_ringshift
  corpus_rotate.each do |text|
    sealed = ringshift_encrypt(text, 17)
    opened = ringshift_decrypt(sealed, 17)
    return false if opened != text
  end
  true
end
